### PR TITLE
bk(smart exhaustive tests): tune the steps to support GH comments and changesets

### PR DESCRIPTION
**Pull request by v1v**
## Release notes
[rn:skip]

## What does this PR do?

Validate the smart exhaustive tests pipeline works as expected

## Why is it important/What is the impact to the user?

Avoid duplicated runs for changes in files and also a valid github comment.
Tune the configuration to run as requested

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- 

## Use cases

Important to highlight that the behaviour of the BK PR bot when running for push and comment events differ, hence the need to have two different implementations when triggering the exhaustive pipeline:
- https://github.com/elastic/logstash/pull/18383#discussion_r2479035592
- https://github.com/elastic/logstash/pull/18383/files/0057a8eabcbf71339e2435c50ee5b5ec58285065#diff-37e558fe06dbd6c97017aa38d26835c3d4516285ee456dcda11c862715392d71R16-R17

In addition, https://github.com/elastic/logstash/pull/18383/files#diff-37e558fe06dbd6c97017aa38d26835c3d4516285ee456dcda11c862715392d71R3 was required, for the same reason, if a GH comment is triggered and the last commit includes changes in the `qa/acceptance` folder, then it would have been triggered twice. That's an edge case I discovered and fixed with the above-mentioned line.

### CODE REVIEW

**Comment by v1v** [comment]
Changes caused https://buildkite.com/elastic/logstash-smart-exhaustive-tests-pipeline/builds/10

While github comment caused https://buildkite.com/elastic/logstash-smart-exhaustive-tests-pipeline/builds/11

File: ci/acceptance_tests.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
+
```
Comment (v1v) [inline ci/acceptance_tests.sh]:
```suggestion
```

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 2]
```diff
@@ -16,19 +16,10 @@ steps:
...
-    build:
-      commit: "${BUILDKITE_COMMIT}"
-      branch: "${BUILDKITE_BRANCH}"
-      env:
-        - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
-        - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
-        - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
-        - ELASTIC_SLACK_NOTIFICATIONS_ENABLE=false
+    if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "/run exhaustive tests"
+    command: "buildkite-agent pipeline upload .buildkite/exhaustive_tests_pipeline.yml"
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml]: Something is not right when using `trigger: "logstash-exhaustive-tests-pipeline"` with the BK PR Bot in this case.

I'll try to debug what's causing the problem

**Comment by v1v** [comment]
#### Test

- [`0057a8e` (#18383)](https://github.com/elastic/logstash/pull/18383/commits/0057a8eabcbf71339e2435c50ee5b5ec58285065) triggered https://buildkite.com/elastic/logstash-smart-exhaustive-tests-pipeline/builds/21
- https://github.com/elastic/logstash/pull/18383#issuecomment-3469342127 triggered https://buildkite.com/elastic/logstash-smart-exhaustive-tests-pipeline/builds/22

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 3]
```diff
@@ -16,19 +16,16 @@ steps:
...
 
   - label: "Trigger logstash-exhaustive-tests-pipeline for GitHub comments"
-    if: build.env("GITHUB_PR_TRIGGER_COMMENT") != ""
+    if: build.env("GITHUB_PR_TRIGGER_COMMENT") == "/run exhaustive tests"
     trigger: "logstash-exhaustive-tests-pipeline"
     build:
-      commit: "${BUILDKITE_COMMIT}"
-      branch: "${BUILDKITE_BRANCH}"
+      commit: "HEAD"
+      branch: "pull/${BUILDKITE_PULL_REQUEST}/merge"
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml:28]: `reviewer`: BK PR bot ehaves awkwardly when supporting GitHub comments, and the previous approach did not work. Please refer to https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/2815

```

2025-10-30 17:36:23 UTC | fatal: reference is not a tree: ecdc1d984499a63372fcb8992c24864c80ab1e5b
-- | --
  | 2025-10-30 17:36:23 UTC | ⚠️ Warning: Checkout failed! checking out commit "ecdc1d984499a63372fcb8992c24864c80ab1e5b": exit status 128 (Attempt 3/3)
  | 2025-10-30 17:36:23 UTC | 🚨 Error: checking out commit "ecdc1d984499a63372fcb8992c24864c80ab1e5b": exit status 128
```

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 steps:
   - label: "Trigger logstash-exhaustive-tests-pipeline for PRs with qa/acceptance/ changes"
-    if: build.pull_request.id != null
+    if: build.pull_request.id != null && build.env("GITHUB_PR_TRIGGER_COMMENT") != "/run exhaustive tests"
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml:3]: in case of two events at the same time, a changeset and also a comment

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 5]
```diff
@@ -16,19 +16,16 @@ steps:
                   commit: "${BUILDKITE_COMMIT}"
                   branch: "${BUILDKITE_BRANCH}"
                   env:
-                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
-                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
-                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
-                    - ELASTIC_SLACK_NOTIFICATIONS_ENABLE=false
+                    BUILDKITE_PULL_REQUEST: "${BUILDKITE_PULL_REQUEST}"
+                    BUILDKITE_PULL_REQUEST_BASE_BRANCH: "${BUILDKITE_PULL_REQUEST_BASE_BRANCH}"
+                    ELASTIC_SLACK_NOTIFICATIONS_ENABLED: false
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml:21]: typo

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 6]
```diff
@@ -16,19 +16,16 @@ steps:
                   commit: "${BUILDKITE_COMMIT}"
                   branch: "${BUILDKITE_BRANCH}"
                   env:
-                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
-                    - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
-                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
-                    - ELASTIC_SLACK_NOTIFICATIONS_ENABLE=false
+                    BUILDKITE_PULL_REQUEST: "${BUILDKITE_PULL_REQUEST}"
+                    BUILDKITE_PULL_REQUEST_BASE_BRANCH: "${BUILDKITE_PULL_REQUEST_BASE_BRANCH}"
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml:20]: `reviewer`: it requires this format, although, I tried to use the same as we have in other places, like `beats` and didn't work.

File: .buildkite/smart_exhaustive_tests_pipeline.yml [hunk 7]
```diff
@@ -16,19 +16,16 @@ steps:
...
-      branch: "${BUILDKITE_BRANCH}"
+      commit: "HEAD"
+      branch: "pull/${BUILDKITE_PULL_REQUEST}/merge"
       env:
-        - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
-        - BUILDKITE_PULL_REQUEST_BASE_BRANCH=${BUILDKITE_PULL_REQUEST_BASE_BRANCH}
-        - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
-        - ELASTIC_SLACK_NOTIFICATIONS_ENABLE=false
+        BUILDKITE_PULL_REQUEST: "${BUILDKITE_PULL_REQUEST}"
+        ELASTIC_SLACK_NOTIFICATIONS_ENABLED: false
```
Comment (v1v) [inline .buildkite/smart_exhaustive_tests_pipeline.yml:31]: typo

**Comment by v1v** [comment]
GH check failed because I cancelled a run related to the changeset by running a GitHub comment immediately after, https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/2820 is running, but the GH check is not updated for some reason, maybe a race condition with the BK PR bot - where cancelled is happening after a new run instead of being sequential :/

File: .buildkite/smart_exhaustive_tests_pipeline.yml (hunk 6 above)
Comment (donoghuc) [reply .buildkite/smart_exhaustive_tests_pipeline.yml:20]: A map of env var name with key and its value as the value makes sense https://github.com/elastic/logstash/blob/be64ebfddc78a987830054a5c277538bda96f014/.buildkite/windows_jdk_matrix_pipeline.yml#L4-L5 over a list of string. Good with this.

File: .buildkite/smart_exhaustive_tests_pipeline.yml (hunk 3 above)
Comment (donoghuc) [reply .buildkite/smart_exhaustive_tests_pipeline.yml:28]: That is odd. Seems like you found a decent workaround though.

**Review by fr4nc1sc0-r4m0n — APPROVED** [review]
LGTM
